src/fprime_gds/flask/static/addons/commanding/arguments.js

```diff
@@ -118,6 +118,26 @@ export function squashify_argument(argument) {
             let field = argument.type.MEMBER_LIST[i][0];
             value[field] = squashify_argument(argument.value[field]);
         }
+    } else if (argument.type.name[0] == 'U') {
```

Comment:
I know this is done the same way lower in the same file, but it seems to me that this logic of checking the first letter of `type.name` to check the type is flawed. 
Say the argument of your command is a `enum` or struct, type.name will carry the value of the C++ type, which will be `ModuleName::EnumName`. Now say you name the module `Foo`, that gets you `Foo::MyEnum`, so the `argument.type.name[0] == 'F'` condition will be true. Which we really don't want.

@LeStarch I feel like we had noticed this before, does that ring a bell? Or was it just me and I forgot to file an issue?

Anyways, since the bug is already there, it's not necessary to fix it in this PR I would say, but we'll want to get it fixed. I'll create an issue.

Reply:
https://github.com/nasa/fprime/issues/2282

Reply:
This was seen before with Floats and structures/Enums that started with `F`.  The correct fix is to track the exact type names [`I8`, `I16`, ..., `U8`, `U16`] and checking against those.

Reply:
The below case is arguably "safe" because `inputType` is only used for specific types (scalar, non-enums) as determined by the Vue template. This case is vulnerable to enums because they are not checked in the `if` statement above (only scalars and arrays are).  Thus enums starting with `U`, `I`, or `F` would hit these statements.

It seems we should write a utility function that checks types against a class type.  Something like:

```
is_scalar_type(type, "signed");
``` 

As this risky check has been used multiple times.

Reply:
Note: declined to write the utility function for now.